### PR TITLE
Move password logic to a service

### PR DESCRIPTION
**Pull request by apardyl**
Refactor, no new features.

### CODE REVIEW

File: src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/UserRegistrationController.kt [hunk 1]
```diff
@@ -47,19 +47,19 @@ class UserRegistrationController(
...
+        if (!user.email.isBlank() && userRepository.findByEmail(user.email) != null) {
             result.rejectValue("email", "email.exists", "email already in use")
         }
 
         if (result.hasErrors()) {
             return "registration/create_account"
         }
 
-        val newUser = User(null, user.userName, null, user.email, user.firstName, user.lastName, false)
+        val newUser = User(null, user.userName, null, user.email, user.firstName, user.lastName, true)
```
Comment (ArcCha) [inline src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/UserRegistrationController.kt]: Constructor without nulls would be better.

File: src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/ExternalUserService.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.Role
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.User
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.UserRepository
+
+@Service
+class ExternalUserService(private val userRepository: UserRepository) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(LdapAttributesMapper::class.java)
```
Comment (ArcCha) [inline src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/ExternalUserService.kt]: I suppose it should be `ExternalUserService::class.java`.

File: src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/UserRegistrationController.kt (hunk 1 above)
Comment (apardyl) [reply src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/UserRegistrationController.kt]: fixed

File: src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/ExternalUserService.kt (hunk 2 above)
Comment (apardyl) [reply src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/ExternalUserService.kt]: fixed

**Pull request merged**